packages/vuetify/src/components/VBadge/VBadge.tsx

```diff
@@ -0,0 +1,176 @@
+// Styles
+import './VBadge.sass'
+
+// Components
+import { VIcon } from '@/components/VIcon'
+
+// Composables
+import { makeTagProps } from '@/composables/tag'
+import { useBackgroundColor, useTextColor } from '@/composables/color'
+import { makeBorderRadiusProps, useBorderRadius } from '@/composables/border-radius'
+import { makeTransitionProps, withTransition } from '@/composables/transition'
+
+// Utilities
+import { computed, defineComponent, toRef } from 'vue'
+import { convertToUnit } from '@/util/helpers'
+
+const extract = (obj: Record<string, unknown>, properties: string[]) => {
+  const extracted: Record<string, unknown> = {}
+  const rest: Record<string, unknown> = {}
+
+  Object.entries(obj).forEach(([key, value]) => {
+    if (properties.includes(key)) {
+      extracted[key] = value
+    } else {
+      rest[key] = value
+    }
+  })
+
+  return [extracted, rest]
+}
+
+export default defineComponent({
+  name: 'VBadge',
+
+  inheritAttrs: false,
+
+  props: {
+    bordered: Boolean,
+    textColor: String,
+    color: {
+      type: String,
+      default: 'primary',
+    },
+    content: {
+      type: String,
+      required: false
+    },
+    dot: Boolean,
+    label: {
+      type: String,
+      default: '$vuetify.badge',
+    },
+    offsetX: [Number, String],
+    offsetY: [Number, String],
+    overlap: {
```

Comment:
I feel boolean true props are a better interface to work with. Maybe a new prop named `floating`?